DAVE2D compilation is broken in v9.2.2

### LVGL version

v9.2.2

### Platform

Board setup: 
- e2studio Version: 2025-01 (25.1.0) Build Id: R20241218-1723
- FSP 5.8.0
- Board EK-RA8D1

Host machine:
- Ubuntu 22.04

### What happened?

While testing the [EK-RA8D1 port](https://github.com/lvgl/lv_port_renesas_ek-ra8d1_gcc), I realized that DAVE2D compilation as some include issues. It can be easily fixed by adding two include lines present in 9.3.

### How to reproduce?

Enable DAV2D support in `lv_conf.h`:

```
/** Use Renesas Dave2D on RA  platforms. */
#define LV_USE_DRAW_DAVE2D 1
```

Then try building the project in e2studio. You end up with a ton of errors (these are the latest one):

```
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:74:23: error: request for member 'p' in something not a structure or union
   74 |             p[2] = dsc->p[1];
      |                       ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:75:23: error: request for member 'p' in something not a structure or union
   75 |             p[1] = dsc->p[0];
      |                       ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:92:31: error: request for member 'bg_grad' in something not a structure or union
   92 |     if(LV_GRAD_DIR_NONE != dsc->bg_grad.dir) {
      |                               ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:104:34: error: request for member 'bg_grad' in something not a structure or union
  104 |         if(LV_GRAD_DIR_VER == dsc->bg_grad.dir) {
      |                                  ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:105:21: error: request for member 'bg_grad' in something not a structure or union
  105 |             a1 = dsc->bg_grad.stops[0].opa;
      |                     ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:106:21: error: request for member 'bg_grad' in something not a structure or union
  106 |             a2 = dsc->bg_grad.stops[dsc->bg_grad.stops_count - 1].opa;
      |                     ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:106:40: error: request for member 'bg_grad' in something not a structure or union
  106 |             a2 = dsc->bg_grad.stops[dsc->bg_grad.stops_count - 1].opa;
      |                                        ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:128:39: error: request for member 'bg_grad' in something not a structure or union
  128 |         else if(LV_GRAD_DIR_HOR == dsc->bg_grad.dir) {
      |                                       ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:133:79: error: request for member 'bg_grad' in something not a structure or union
  133 |         d2_setcolor(u->d2_handle, 0, lv_draw_dave2d_lv_colour_to_d2_colour(dsc->bg_grad.stops[0].color));
      |                                                                               ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:137:38: error: request for member 'bg_opa' in something not a structure or union
  137 |         d2_setalpha(u->d2_handle, dsc->bg_opa);
      |                                      ^~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c:139:79: error: request for member 'bg_color' in something not a structure or union
  139 |         d2_setcolor(u->d2_handle, 0, lv_draw_dave2d_lv_colour_to_d2_colour(dsc->bg_color));
      |                                                                               ^~
make: *** [src/lvgl/src/draw/renesas/dave2d/subdir.mk:52: src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.o] Error 1
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_label.c:15:66: error: 'lv_draw_dave2d_draw_letter_cb' undeclared (first use in this function)
   15 |     lv_draw_label_iterate_characters(&u->base_unit, dsc, coords, lv_draw_dave2d_draw_letter_cb);
      |                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_label.c:15:66: note: each undeclared identifier is reported only once for each function it appears in
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_label.c: At top level:
../src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_label.c:20:43: error: unknown type name 'lv_draw_fill_dsc_t'; did you mean 'lv_draw_glyph_dsc_t'?
   20 |                                           lv_draw_fill_dsc_t * fill_draw_dsc, const lv_area_t * fill_area)
      |                                           ^~~~~~~~~~~~~~~~~~
      |                                           lv_draw_glyph_dsc_t
make: *** [src/lvgl/src/draw/renesas/dave2d/subdir.mk:52: src/lvgl/src/draw/renesas/dave2d/lv_draw_dave2d_label.o] Error 1
```

--- a/Kconfig
+++ b/Kconfig
@@ -1,4 +1,4 @@
-# Kconfig file for LVGL v9.2.2
+# Kconfig file for LVGL v9.2.3
 
 menu "LVGL configuration"
 
@@ -1492,7 +1492,7 @@ menu "LVGL configuration"
 			default 2 # LVGL_VERSION_MINOR
 		config LVGL_VERSION_PATCH
 			int
-			default 2 # LVGL_VERSION_PATCH
+			default 3 # LVGL_VERSION_PATCH
 	endmenu
 
 	menu "Devices"

--- a/env_support/cmake/version.cmake
+++ b/env_support/cmake/version.cmake
@@ -1,7 +1,7 @@
 set(LVGL_VERSION_MAJOR "9")
 set(LVGL_VERSION_MINOR "2")
-set(LVGL_VERSION_PATCH "2")
-set(LVGL_VERSION_INFO  "")
+set(LVGL_VERSION_PATCH "3")
+set(LVGL_VERSION_INFO  "dev")
 
 set(LVGL_VERSION ${LVGL_VERSION_MAJOR}.${LVGL_VERSION_MINOR}.${LVGL_VERSION_PATCH}) # This is a CMake variable
 set(ENV{LVGL_VERSION} ${LVGL_VERSION}) # This is exported Environmental variable

--- a/library.json
+++ b/library.json
@@ -1,6 +1,6 @@
 {
 	"name": "lvgl",
-	"version": "9.2.2",
+	"version": "9.2.3",
 	"keywords": "graphics, gui, embedded, tft, lvgl",
 	"description": "Graphics library to create embedded GUI with easy-to-use graphical elements, beautiful visual effects and low memory footprint. It offers anti-aliasing, opacity, and animations using only one frame buffer.",
 	"repository": {

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=lvgl
-version=9.2.2
+version=9.2.3
 author=kisvegabor
 maintainer=kisvegabor,embeddedt,pete-pjb
 sentence=Full-featured Graphics Library for Embedded Systems

--- a/lv_conf_template.h
+++ b/lv_conf_template.h
@@ -1,6 +1,6 @@
 ﻿/**
  * @file lv_conf.h
- * Configuration file for v9.2.2
+ * Configuration file for v9.2.3-dev
  */
 
 /*

--- a/lv_version.h
+++ b/lv_version.h
@@ -8,7 +8,7 @@
 
 #define LVGL_VERSION_MAJOR 9
 #define LVGL_VERSION_MINOR 2
-#define LVGL_VERSION_PATCH 2
-#define LVGL_VERSION_INFO ""
+#define LVGL_VERSION_PATCH 3
+#define LVGL_VERSION_INFO "dev"
 
 #endif /* LVGL_VERSION_H */

--- a/src/draw/renesas/dave2d/lv_draw_dave2d.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d.c
@@ -9,6 +9,7 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 #include "../../lv_draw_buf_private.h"
+#include "../../../misc/lv_area_private.h"
 
 /*********************
  *      DEFINES

--- a/src/draw/renesas/dave2d/lv_draw_dave2d.h
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d.h
@@ -13,11 +13,20 @@ extern "C" {
 /*********************
  *      INCLUDES
  *********************/
+
 #include "../../../lv_conf_internal.h"
 #if LV_USE_DRAW_DAVE2D
 #include "../../lv_draw.h"
+#include "../../lv_draw_private.h"
 #include "hal_data.h"
 #include "lv_draw_dave2d_utils.h"
+#include "../../lv_draw_rect.h"
+#include "../../lv_draw_line.h"
+#include "../../lv_draw_arc.h"
+#include "../../lv_draw_label.h"
+#include "../../lv_draw_image.h"
+#include "../../lv_draw_triangle.h"
+#include "../../lv_draw_buf.h"
 
 /*********************
  *      DEFINES

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_arc.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_arc.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 void lv_draw_dave2d_arc(lv_draw_dave2d_unit_t * u, const lv_draw_arc_dsc_t * dsc, const lv_area_t * coords)
 {
 

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_border.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_border.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 static void dave2d_draw_border_complex(lv_draw_dave2d_unit_t * draw_unit, const lv_area_t * outer_area,
                                        const lv_area_t * inner_area,
                                        int32_t rout, int32_t rin, lv_color_t color, lv_opa_t opa);

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_fill.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_fill.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 void lv_draw_dave2d_fill(lv_draw_dave2d_unit_t * u, const lv_draw_fill_dsc_t * dsc, const lv_area_t * coords)
 {
     lv_area_t draw_area;

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_image.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_image.c
@@ -9,6 +9,9 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../lv_image_decoder_private.h"
+#include "../../lv_draw_image_private.h"
+
 /*********************
  *      DEFINES
  *********************/

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_label.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_label.c
@@ -1,6 +1,9 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../lv_draw_label_private.h"
+#include "../../../misc/lv_area_private.h"
+
 static void lv_draw_dave2d_draw_letter_cb(lv_draw_unit_t * draw_unit, lv_draw_glyph_dsc_t * glyph_draw_dsc,
                                           lv_draw_fill_dsc_t * fill_draw_dsc, const lv_area_t * fill_area);
 

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_line.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_line.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 void lv_draw_dave2d_line(lv_draw_dave2d_unit_t * u, const lv_draw_line_dsc_t * dsc)
 {
 

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_mask_rectangle.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_mask_rectangle.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 void lv_draw_dave2d_mask_rect(lv_draw_dave2d_unit_t * u, const lv_draw_mask_rect_dsc_t * dsc, const lv_area_t * coords)
 {
     lv_area_t clipped_area;

--- a/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c
+++ b/src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c
@@ -1,6 +1,8 @@
 #include "lv_draw_dave2d.h"
 #if LV_USE_DRAW_DAVE2D
 
+#include "../../../misc/lv_area_private.h"
+
 void lv_draw_dave2d_triangle(lv_draw_dave2d_unit_t * u, const lv_draw_triangle_dsc_t * dsc)
 {
     lv_area_t clipped_area;

